src/track/track.cpp

```diff
@@ -311,7 +311,7 @@ QString Track::getBpmText() const {
 
 void Track::setBeats(mixxx::BeatsPointer pBeats) {
     QMutexLocker lock(&m_qMutex);
-    setBeatsMarkDirtyAndUnlock(&lock, pBeats);
+    setBeatsMarkDirtyAndUnlock(&lock, pBeats, false);
```

Comment:
Same here, `m_record.getBpmLocked()` instead of `false`, the bpm lock should be unaffected.